### PR TITLE
fix(firebase_messaging, web): Define scope for 'firebase-messaging-sw.js' on registration

### PR DESCRIPTION
**Pull request by AnayShri**
## Description

> fixes: #12638

Add scope for '**firebase-messaging-sw.js**' on registration itself. The scope remains unchanged from the existing scope of the Firebase messaging service worker, only the definition is now at registration as compared to earlier behaviour which happens after initialization.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: tests/web/index.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
-<head>
-  <!--
+  <head>
```
Comment (TarekkMA) [inline tests/web/index.html:3]: Can you revert formatting for this file, only keep changed lines of code

**Review by TarekkMA — CHANGES_REQUESTED** [review]
Thank you @AnayShri  for your PR.

File: tests/web/index.html (hunk 1 above)
Comment (AnayShri) [reply tests/web/index.html:3]: I guess it auto-formatted on push, will revert the auto-formatting.

**Review by russellwheatley — APPROVED** [review]

**Comment by russellwheatley** [comment]
Default scope for messaging: https://github.com/firebase/firebase-js-sdk/blob/b74d8a2115b757d73bd7bf6dba27f9516b380366/packages/messaging/src/util/constants.ts#L19

**Review by Lyokone — APPROVED** [review]

**Review by TarekkMA — APPROVED** [review]

**Pull request merged**